src/features/projectsV2/ProjectsMap/index.tsx

```diff
@@ -205,37 +212,58 @@ function ProjectsMap(props: ProjectsMapProps) {
     },
     [plantLocations, props.page, selectedPlantLocation]
   );
-
+  /**
+   * This onClick handler is responsible for:
+   * - Selecting: point plant locations(single tree), polygon plant locations(multi tree), or project sites
+   * - Deselecting: point plant locations ,sample point plant locations, other interventions(point geometry)
+   */
   const onClick = useCallback(
     (e) => {
       if (props.page !== 'project-details') return;
-      const hasNoSites = singleProject?.sites?.length === 0;
-      const result = getPlantLocationInfo(plantLocations, mapRef, e.point);
-      const isSamePlantLocation =
-        result?.geometry.type === 'Point' &&
-        result.id === selectedPlantLocation?.id;
-      const isSingleTree =
-        selectedPlantLocation?.type === 'single-tree-registration';
-      const isMultiTree =
-        selectedPlantLocation?.type === 'multi-tree-registration';
-      // const isOther = selectedPlantLocation?.type !== 'single-tree-registration' && selectedPlantLocation?.type !== 'multi-tree-registration';
-      // Clear sample plant location on clicking outside.
-      // Clicks on sample plant location will not propagate on the map
-      setSelectedSamplePlantLocation(null);
-      // Clear plant location info if clicked twice (single or multi tree) // point plant location
-      if (isSamePlantLocation && (isSingleTree || isMultiTree)) {
+
+      const features = getFeaturesAtPoint(mapRef, e.point);
+      if (!features || features.length === 0) return;
+
+      const plantLocationInfo = getPlantLocationInfo(plantLocations, features);
+      const siteIndex = getSiteIndex(singleProject?.sites || [], features);
+
+      const isSamePlant = plantLocationInfo?.id === selectedPlantLocation?.id;
+      const isPointGeometry =
+        plantLocationInfo &&
+        (plantLocationInfo.geometry.type === 'Point' ||
+          otherInterventionsWithPointGeometry.includes(plantLocationInfo.type));
```

Comment:
Not sure why we need the condition `otherInterventionsWithPointGeometry.includes(plantLocationInfo.type)` - I have 2 concerns about this:

1. Are these the only interventions that can have point geometry?
2. We could have a case where the plant location is not a point, but this condition holds true. e.g. A polygon plant location of type `fire-patrol`.